### DISCUSSION TITLE
I fixed my pre-1.6.1 theme by adding the `-no-default-config` option

### DISCUSSION THREAD

**Discussion by ouuan** [davatorium/rofi#1408] [Theming]
I'm using a slightly modified version of https://github.com/MonsieurBedard/rofi-breeze-theme.

This is the normal appearance:

![normal](https://user-images.githubusercontent.com/30581822/131460730-a5cffba3-484c-4453-bed7-f1f718dca05d.png)

This is the appearance in rofi 1.7.0 without `-no-default-config`:

![strange](https://user-images.githubusercontent.com/30581822/131460682-99b4a118-6bc5-4b7e-a417-2f1749366712.png)

And everything is normal again when I add `-no-default-config` in the rofi options.

Hope this will be helpful to others.

**Comment by DaveDavenport** [comment]
### Please do not use this.

In your config.rasi include your theme using:
```
@theme "mytheme"
```
This should get the same result, or you can fix the theme.

Not including the default configuration will have undesired side effects.

**Reply by DaveDavenport** [reply]
If you use:
```
@import
```
It will APPEND to the existing theme

```
@theme 
```
will clear the existing theme and load the new one.

See the rofi-theme manpage

**Reply by HumblePresent** [reply]
So if I have my theme specified in `config.rasi` I should move it to a separate file and use `@theme`? I'm a bit confused on why my theme settings specified in `config.rasi` are being overwritten by the default theme.

**Reply by DaveDavenport** [reply]
its not overwritten. 

By default  it loads the default theme,  you can **add** items to the default  theme by adding them to your config.rasi.
This is so you can do simple tweaks to the default theme (or if after an `@theme` slightly tweak a pre-installed theme).

The intention is to do
```css
@theme "mytheme"
```
in your config loading the the actual theme.
(that can use @import if it is split over multiple files.)

tdlr;  Adding theme tweaks to your config.rasi is for tweaking themes,  to load a fresh theme (discarding the default theme) use `@theme`.

**Reply by HumblePresent** [reply]
Ok that makes sense. It still seems like some setting don't take affect though if specified in `config.rasi`. For example
```
element {
    background-color: #282828;
}
```
I expect this to modify the default theme by changing the background color of the elements in listview but when I run they still have the default background color. Maybe this is a more specific issue and I should create my own discussion but this seems like a simple test case that should work.

**Reply by DaveDavenport** [reply]
You might want to take a look at the default theme first before making assumptions. The default theme is explicit for all variations, so just setting `element {}` does not do anything visible.
(`rofi -no-config -dump-theme`)

**Reply by DaveDavenport** [reply]
Try f.e:
```bash
rofi -no-config -theme-str "element normal.normal  { background-color: blue;}" -show run
```

**Reply by DaveDavenport** [reply]
Also check out the `rofi-theme` manpage.

**Reply by HumblePresent** [reply]
Yeah I think I understand now thanks.

**Reply by DaveDavenport** [reply]
I hope this will in the end be a lot more logical and consistent then the pre 1.7.0 release that had tons of internal hacks trying to make 3+ versions of themes working.

**Reply by HumblePresent** [reply]
Yeah ripping off the bandaid hurts at first but once everyone fixes their themes future updates will be easier

**Comment by ecesar88** [comment]
I'm still very confused on how I should fix my theme. Even using the `-dump-config` option dumping it to a file and trying to use that isn't working for me, the same happens with the `-dump-theme` option. I think i'll just go back to using dmenu, because it is sincerely taking a lot of my time to configure back to it's working state (pre-1.6.1).

**Reply by DaveDavenport** [reply]
Not sure what you want to achieve with this post?   without more information it is hard to provide support.
Maybe if you pastebin your old setup we might be able to help.

**Reply by ecesar88** [reply]
Yep, you're right! I just opened an discussion including more details on the problem: https://github.com/davatorium/rofi/discussions/1491
